src/renderer/components/maps/MapBattlePreviewStartBox.vue

```diff
@@ -0,0 +1,473 @@
+        resetDrag();
+        return;
+    }
+
+    // Get the final position from the element's style
+    const left = parseFloat(boxElement.value.style.left) / 100;
+    const top = parseFloat(boxElement.value.style.top) / 100;
+
+    // Create a new box with updated position but same size
+    const newBox = {
+        left,
+        top,
+        right: left + (startBox.value.right - startBox.value.left),
+        bottom: top + (startBox.value.bottom - startBox.value.top),
+    };
+
+    // Clear inline styles
+    boxElement.value.style.left = "";
+    boxElement.value.style.top = "";
+    boxElement.value.classList.remove("dragging");
+
+    // Emit the update
+    updateBoxInStore(newBox);
+
+    resetDrag();
+}
+
+function resetDrag() {
+    isDragging.value = false;
+    startBox.value = null;
+    document.removeEventListener("mousemove", handleDrag);
+    document.removeEventListener("mouseup", endDrag);
+}
+
+// Resize handlers
+function startResize(side: string | null, corner: string | null, event: MouseEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+    isResizing.value = true;
+
+    if (side) activeHandle.value = side;
+    if (corner) activeHandle.value = corner;
+
+    // Store initial positions
+    startPos.value = { x: event.clientX, y: event.clientY };
+    startBox.value = { ...props.box };
+
+    // Get the parent container dimensions
+    const mapElement = boxElement.value?.closest(".map");
+    if (mapElement) {
+        parentRect.value = mapElement.getBoundingClientRect();
+    }
+
+    // Add event listeners for move and up
+    document.addEventListener("mousemove", handleResize);
+    document.addEventListener("mouseup", endResize);
+
+    // Add visual feedback
+    if (boxElement.value) {
+        boxElement.value.classList.add("resizing");
+    }
+
+    // Set cursor for the entire document during resize
+    if (side === "n-side" || side === "s-side") {
+        document.body.style.cursor = "ns-resize";
+    } else if (side === "e-side" || side === "w-side") {
+        document.body.style.cursor = "ew-resize";
+    } else if (corner === "ne-corner" || corner === "sw-corner") {
+        document.body.style.cursor = "nesw-resize";
+    } else if (corner === "nw-corner" || corner === "se-corner") {
+        document.body.style.cursor = "nwse-resize";
+    }
+}
+
+function handleResize(event: MouseEvent) {
+    if (!isResizing.value || !startBox.value || !parentRect.value || !activeHandle.value || !boxElement.value) return;
+
+    // Calculate the movement in normalized coordinates (0-1)
+    const dx = (event.clientX - startPos.value.x) / parentRect.value.width;
+    const dy = (event.clientY - startPos.value.y) / parentRect.value.height;
+
+    // Start with the initial size
+    const newBox = { ...startBox.value };
+
+    // Update the appropriate sides based on which handle is dragged
+    switch (activeHandle.value) {
```

Comment:
Optional:

If active handle was ```{v: 'n' | 's' | null, h: 'w' | 'e' | null}``` then:

In the switch below it would be:

```js
switch (activeHandle.h) {
  case 'w':
    newBox.left = Math.min(newBox.right - 0.05, startBox.value.left + dx);
    break;
  case 'e':
    newBox.right = Math.max(newBox.left + 0.05, startBox.value.right + dx);
    break;
}
switch (activeHandle.v) {
  case 'n':
    newBox.top = Math.min(newBox.bottom - 0.05, startBox.value.top + dy);
    break;
  case 's':
    newBox.bottom = Math.max(newBox.top + 0.05, startBox.value.bottom + dy);
    break;
}
```

Above `startResize` could just take that struct as argument or take `v` + `h` (vertical, horizontal)